### ISSUE TITLE
"Alias not found" although exposed and given as direct dependency in Elm

### ISSUE THREAD

**Issue by ogreyard** [dillonkearns/elm-typescript-interop#21]
Hi Dillon,

first of all, thanks for the package, I have come across an issue that refuses to be solved by myself alone, so I'd appreciate your help. 
I am running into the error message 

`Alias 'Spec' not found. Known aliases: Main.Model`

`Spec` is a type alias from https://package.elm-lang.org/packages/gicentre/elm-vega/latest/Vega and I have imported and exposed it in my Main.elm as follows

`import Vega exposing (Spec)`

I can compile my Main.elm to Main.js without any issue, however the 

`npx elm-typescript-interop` gives the said error message and the .d.ts file is not created.
What am I missing?

Thanks for your help.

Kind regards
Markus

**Comment by dillonkearns** [comment]
Hello Markus,

It looks like there are a couple of things going on here.

First, I'm not currently pulling in the source code for the dependencies in a project, so if you want to use a type alias that's defined in an external package you would need to workaround it by manually copying that type alias into your own project's source.

However, it looks like in this particular case, `type alias Spec = Value`. And [Value is defined](https://package.elm-lang.org/packages/gicentre/elm-vega/latest/Vega#Value) as an elm Custom Type, which means that it's not a primitive.

Elm only allows the following kinds of values to be sent directly through ports:

- Booleans and Strings
- Numbers
- Lists
- Arrays
- Tuples
- Records
- Maybes
- Json.Encode.Value/Json.Decode.Value

And `elm-typescript-interop` doesn't change the elm language, it just generates a TypeScript definition file for your elm code, but it must be valid elm.

I might be missing something here, so if that's the case then you can use the workaround I mentioned in the first part of the message. But otherwise, what you'll need to do is define your own function for turning a `Vega.Value` into one of the primitives listed above. Then you can send that value directly using a port, and elm-typescript-interop will annotate it appropriately.

I hope that helps! Let me know if you have any other questions, I'm happy to point you in the right direction.

**Comment by ogreyard** [comment]
Hello Dillon, 

first of all, thanks for your very timely answer and the detailed explanations!

When I copied the `type alias Spec = Value` into my Main.elm, it was in fact giving me the error that ports can only transfer primitives and others as defined above in your answer.

However from the tutorial shown here https://github.com/gicentre/elm-vega/tree/master/docs/helloWorld I figured it must be possible and it indeed worked before like that. That's why I looked at the [source](https://github.com/gicentre/elm-vega/blob/ed047b8c8e41e92e2128d6f5d50f4a3396043330/src/Vega.elm#L4194) where it states 

`type alias Spec = JE.Value` 

with

`import Json.Encode as JE`

which makes a lot of sense as then the Spec is just a JSON.

Copying the type alias like 

`type alias Spec = Json.Encode.Value` solves the port error and also made the elm-typescript-interop command work again.

Thanks a lot!
